amazonclouddrive/amazonclouddrive.go

```diff
@@ -597,6 +597,110 @@ func (f *Fs) Mkdir() error {
 	return f.dirCache.FindRoot(true)
 }
 
+// Move src to this remote using server side move operations.
+//
+// This is stored with the remote path given
+//
+// It returns the destination Object and a possible error
+//
+// Will only be called if src.Fs().Name() == f.Name()
+//
+// If it isn't possible then return fs.ErrorCantMove
+func (f *Fs) Move(src fs.Object, remote string) (fs.Object, error) {
+	srcObj, ok := src.(*Object)
+	if !ok {
+		fs.Debug(src, "Can't move - not same remote type")
+		return nil, fs.ErrorCantMove
+	}
+	fs.Debug(src, "\n\nTrying to move '%s' -> '%s'", srcObj.Remote(), remote)
+
+	// Temporary Object under construction
+	dstObj := &Object{
+		fs:     f,
+		remote: remote,
+	}
+	// Check if object already exists
+	err := dstObj.readMetaData()
+	switch err {
+	case nil:
+		// TODO: directory exists?
+		fs.Debug(src, "Move target '%s' already exists, removing it first.")
+		err := dstObj.Remove()
+		if err != nil {
+			return nil, err
+		}
+		dstObj.info = nil
+	case fs.ErrorObjectNotFound:
+		// Not found so we can move it there
+	default:
+		return nil, err
+	}
+
+	sLeaf, sDirectoryID, err := f.dirCache.FindPath(srcObj.Remote(), false)
+	if err != nil {
+		return nil, err
+	}
+	tLeaf, tDirectoryID, err := f.dirCache.FindPath(remote, true)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(srcObj.info.Parents) > 1 && sLeaf != tLeaf {
+		fs.Debug(src, "Cannot move, because object is attached to multiple parents and should be renamed. This would change the name of the node in all parents.")
+		return nil, fs.ErrorCantMove
+	}
+
+	if sLeaf != tLeaf {
+		fs.Debug(src, "renaming")
+		err = srcObj.Rename(tLeaf)
+		if err != nil {
+			fs.Debug(src, "Move: quick path rename failed: %v", err)
+			goto OnConflict
+		}
+	}
+	if sDirectoryID != tDirectoryID {
+		err = srcObj.ReplaceParent(sDirectoryID, tDirectoryID)
+		if err != nil {
+			fs.Debug(src, "Move: quick path parent replace failed: %v", err)
+			return nil, err
+		}
+	}
+	return dstObj, nil
+
+OnConflict:
+	fs.Debug(src, "OMG WE HAD A CONFLICT, DOING SUPER COMPLICATED METHOD")
```

Comment:
:smile: 

Can you write a comment here explaining a) why the simple method above didn't work, and b) what the super complicated method does - thanks!